lib/service_contract_webmock/service_mock.rb

```diff
@@ -55,11 +63,46 @@ def stub_search_request
       matcher = ContractMatcher.new(contract.endpoint("index"), resources)
       stub_request(:get, "#{name}\\.json\\?#{matcher.to_regex}").
         to_return do |request|
+          found = matcher.found(request.uri.query)
           {
-            body: { name => matcher.found(request.uri.query) }.to_json,
+            body: {
+              name => apply_pagination(request, found),
+              meta: pagination_meta(request, found).merge({ status: 200})
+            }.to_json,
             headers: { 'Content-Type' => 'application/json' }
           }
         end
     end
+
+    private
+    def apply_pagination(request, resources)
+      pagination_params = extract_pagination_params(request.uri.query, resources)
+      per_page = pagination_params["per_page"].to_i
+      current_page = pagination_params["page"].to_i
+
+      resources.slice(((current_page - 1) * per_page), per_page) || []  # Array#slice returns nil when index is out of bounds
+    end
+
+    def pagination_meta(request, resources)
+      pagination_params = extract_pagination_params(request.uri.query, resources)
+      per_page = pagination_params["per_page"].to_i
+      total_pages = resources.count / per_page
+      total_pages += 1 if resources.count % per_page > 0
+
+      {
+        current_page: pagination_params["page"].to_i,
+        per_page: per_page,
+        total_pages: total_pages,
+        total_entries: resources.count
+      }
+    end
+
+    def extract_pagination_params(query, resources)
+      params = Rack::Utils.parse_nested_query(query).slice("page", "per_page")
+      params["page"] = 1 if params["page"].nil? || params["page"].empty?
+      params["per_page"] = [resources.count, 1].max if params["per_page"].nil? || params["per_page"].empty?
```

Comment:
Having `per_page` set to `resources.count` makes it compatible with other non-paginated implemetations